### PR TITLE
update github actions to use tf-actions

### PR DESCRIPTION
**Pull request by cmharlow**
Jira: https://mozilla-hub.atlassian.net/browse/OPST-113

What this PR does:
* Setups Terraform CI to use https://github.com/mozilla/tf-actions, a product of the GCP Migration / Terraform Standardization working groups

(recreated this work on a new branch to test an upstream fix)

### CODE REVIEW

**Review by smarnach — COMMENTED** [review]
I'm lacking a bit of context here to usefully review this, so I just have a couple of questions.

* It looks like the "matrix" generated by the matrixify job is empty for this PR. Is this because the PR does not change code in any Terraform module? Can we somehoew test this in a situation where the CI job isn't skipped, just to see that it is working as intended?

* Could the two jobs here be combined into a single combined action, so clients of tf-actions need less boilerplate code?

**Comment by cmharlow** [comment]
Alright, questions from @smarnach were answered in a pairing session, and I've added updates to the modules we used for testing the CI to get CI to pass and demo some of the new expected structure.

If this PR looks okay, this will get baseline (stateless) CI setup. Then what I propose for @rvaralda and others is I can make a list of tickets for reviewing each module here for:
1. do we still use / need this module?
2. if yes to 1, update module's structure, expected terraform version, and fix tagging duplication bug (ask Christina for more info on this if you need it)
3. once through 1 & 2 for all modules, migrate the cleaned up & reviewed modules into github.com/mozilla/terraform-modules for sharing across SRE team & update our clients to use that repository

File: aws/database/.terraform.lock.hcl [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This file is maintained automatically by "terraform init".
```
Comment (smarnach) [inline aws/database/.terraform.lock.hcl:1]: Does it make sense to have this file here? I thought it's only useful for `terraform init`, but we don't run `terraform init` in a module.

File: gcp/identity/.terraform.lock.hcl [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# This file is maintained automatically by "terraform init".
```
Comment (smarnach) [inline gcp/identity/.terraform.lock.hcl:1]: Same question here.

**Review by smarnach — COMMENTED** [review]
It would be great if there was some way of forcing a run for all modules, rather than just the changed modules. Maybe we can have a manually triggered workflow that tests everything?

File: .gitignore [hunk 3]
```diff
@@ -1 +1,9 @@
-.terraform
+# Local .terraform directories
+**/.terraform
```
[the remark this answers is not in the document]
Comment (cmharlow) [reply .gitignore:2]: i think it explicitly covers child directory .terraform files.

Can include both if you want though.

File: .gitignore [hunk 4]
```diff
@@ -1 +1,9 @@
...
+
+# .tfstate files
+*.tfstate
+*.tfstate.*
```
[the remark this answers is not in the document]
Comment (cmharlow) [reply .gitignore:6]: yes, though we have a few examples (early versions of websre-infra) where state was stored in repository / on local filesystem, so making explicit.

File: .gitignore [hunk 5]
```diff
@@ -1 +1,9 @@
...
+
+# .tfvars files
+*.tfvars
```
[the remark this answers is not in the document]
Comment (cmharlow) [reply .gitignore:9]: fair. practices could be to use tfvars with secret values, but no indication we're trying to support that workflow here. removing.

File: aws/database/.terraform.lock.hcl (hunk 1 above)
Comment (cmharlow) [reply aws/database/.terraform.lock.hcl:1]: Yeah so this came up elsewhere, but basically, you can run terraform validate on a module, but it requires an init, hence this file. from an earlier mozilla/terraform-modules PR on this:

```
tf init & tf validate in module directory creates the .terraform.lock.hcl file... which i wasn't sure if we wanted to keep in modules specifically (it'll get ignored in usage in terraform projects as I understand it). this Hashi discussion led me to checking it in though
```

so I think we're good to commit these init lock files in modules where such CI may occur, but its not harmful to leave out

File: gcp/identity/.terraform.lock.hcl (hunk 2 above)
Comment (cmharlow) [reply gcp/identity/.terraform.lock.hcl:1]: Same answer as above.

File: aws/database/.terraform.lock.hcl (hunk 1 above)
Comment (cmharlow) [reply aws/database/.terraform.lock.hcl:1]: this is the link: https://discuss.hashicorp.com/t/what-is-the-best-practise-with-terraform-lock-hcl-in-modules-commit-or-not/28648

**Comment by cmharlow** [comment]
> It would be great if there was some way of forcing a run for all modules, rather than just the changed modules. Maybe we can have a manually triggered workflow that tests everything?

feel free to open a feature request on this in https://mozilla-hub.atlassian.net/browse/OPST-232 but we don't have support for this currently.

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (smarnach) [reply .gitignore:2]: Just `.terraform` matches `.terraform` in this directory or any subdirectory, so I believe they are the same. I don't care whether it's one or the other – I just wanted to understand why you changed it.

File: aws/database/.terraform.lock.hcl (hunk 1 above)
Comment (smarnach) [reply aws/database/.terraform.lock.hcl:1]: So this basically fixes the versions of dependencies that are used when running `terraform validate`. Including the file here has the advantage of having more reproducible builds, but the downside that we need to update the file every now and then, and that we might catch some errors later.

I don't really have an opinion either way, so let's just keep it here.

**Review by smarnach — APPROVED** [review]

**Comment by cmharlow** [comment]
Added ticket for request here: https://mozilla-hub.atlassian.net/browse/OPST-290

**Pull request merged**